examples/l2switch/l2switch.c

```diff
@@ -0,0 +1,326 @@
+                rte_eth_macaddr_get(ports -> id[i], &l2fwd_ports_eth_addr[ports -> id[i]]);
+		printf("Port %u, MAC address: %02X:%02X:%02X:%02X:%02X:%02X\n\n",
+                        ports ->id[i],
+                        l2fwd_ports_eth_addr[ports -> id[i]].addr_bytes[0],
+                        l2fwd_ports_eth_addr[ports -> id[i]].addr_bytes[1],
+                        l2fwd_ports_eth_addr[ports -> id[i]].addr_bytes[2],
+                        l2fwd_ports_eth_addr[ports -> id[i]].addr_bytes[3],
+                        l2fwd_ports_eth_addr[ports -> id[i]].addr_bytes[4],
+                        l2fwd_ports_eth_addr[ports -> id[i]].addr_bytes[5]);
+        }
+}
+
+/* The source MAC address is replaced by the TX_PORT MAC address */
+/* The destination MAC address is replaced by 02:00:00:00:00:TX_PORT_ID */
+static void
+l2fwd_mac_updating(struct rte_mbuf *pkt, unsigned dest_portid) {
+	struct ether_hdr *eth;
+	void *tmp;
+
+	eth = rte_pktmbuf_mtod(pkt, struct ether_hdr *);
+
+	/* 02:00:00:00:00:xx */
+	tmp = &eth->d_addr.addr_bytes[0];
+	*((uint64_t *)tmp) = 0x000000000002 + ((uint64_t)dest_portid << 40);
+
+	ether_addr_copy(tmp, &eth->s_addr);
+
+        if (print_mac) {
+                printf("Packet updated MAC address: %02X:%02X:%02X:%02X:%02X:%02X\n\n",
+                        eth->s_addr.addr_bytes[0],
+                        eth->s_addr.addr_bytes[1],
+                        eth->s_addr.addr_bytes[2],
+                        eth->s_addr.addr_bytes[3],
+                        eth->s_addr.addr_bytes[4],
+                        eth->s_addr.addr_bytes[5]);
+        }
+}
+
+/* The destination port is the adjacent port from the enabled portmask, that is, 
+ * if the first four ports are enabled (portmask 0xf),
+ * ports 1 and 2 forward into each other, and ports 3 and 4 forward into each other.
+*/
+static void
+l2fwd_set_dest_ports() {
+        int i;
+        unsigned nb_ports_in_mask = 0;
+        int last_port = 0;
+        for (i = 0; i < ports -> num_ports; i++) {
+                if (nb_ports_in_mask % 2) {
+                        l2fwd_dst_ports[ports -> id[i]] = last_port;
+                        l2fwd_dst_ports[last_port] = ports -> id[i];
+                } else {
+                        last_port = ports -> id[i];
+                }
+                nb_ports_in_mask++;
+        }
+        if (nb_ports_in_mask % 2) {
+                        printf("Notice: odd number of ports in portmask.\n");
+                        l2fwd_dst_ports[last_port] = last_port;
+        }
+}
+static int
+packet_handler(struct rte_mbuf *pkt, struct onvm_pkt_meta *meta,
+               __attribute__((unused)) struct onvm_nf_local_ctx *nf_local_ctx) {
+        static uint32_t counter = 0;
+        if (counter++ == print_delay) {
+                print_stats();
+                counter = 0;
+        }
+        /* Update stats packet received on port. */
+        port_statistics[pkt -> port].rx += 1;
```

Comment:
If pkt->port is greater than RTE_ETH_MAXPORTS then we can get a seg fault. Add a check in here that makes sure it is within the bounds